pkg/module/disk_ref.go

```diff
@@ -20,6 +21,29 @@ type diskRef struct {
 	version string
 }
 
+type zipReadCloser struct {
+	zip io.ReadCloser
+	ref *diskRef
+}
+
+// Close closes the zip file handle and clears up disk space used by the underlying disk ref
+// It is the caller's responsibility to call this method to free up utilized disk space
+func (rc *zipReadCloser) Close() error {
+	rc.zip.Close()
+	return clearFiles(rc.ref.fs, rc.ref.root)
+}
+
+func (rc *zipReadCloser) Read(p []byte) (n int, err error) {
+	return rc.zip.Read(p)
+}
+
+func newZipReadCloser(zip io.ReadCloser, ref *diskRef) io.ReadCloser {
```

Comment:
you can remove this function and just inline below.

Reply:
yeah that makes sense, that was unnecessarily verbose :)

Thanks for the review!